### PR TITLE
[GLUTEN-4827][UT] Add Golden Files for TPC-H Spark34 + Gluten Execution Plan

### LINKED ISSUE #4827: [UT] Add Golden Files for TPC-H Spark34 + Gluten Execution Plan
### Description

As title.

### PR DESCRIPTION
**Pull request by zwangsheng**
## What changes were proposed in this pull request?

Close #4827

As title, follow up to add TPC-H + Spark33 Golden Files.

## How was this patch tested?

unit tests

### CODE REVIEW

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -308,6 +308,21 @@ jobs:
...
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: golden-files-spark34
+          path: |
+            /tmp/${{ github.run_id }}/spark34/tpch-approved-plan/**
+      - name: Clean temp golden files
+        if: failure()
+        run: |
+          rm -rf /tmp/$GITHUB_RUN_ID/spark34/tpch-approved-plan
```
Comment (PHILO-HE) [inline .github/workflows/velox_be.yml:325]: Can this be moved to the above "Upload golden files"  GHA task ? If feasible, maybe we can unify them (the above two or three tasks) into one task.

Comment (zwangsheng) [reply .github/workflows/velox_be.yml:325]: I got your point, i will go futher to find how to resolve this.

File: .github/workflows/velox_be.yml [hunk 2]
```diff
@@ -631,3 +616,17 @@ jobs:
         if: ${{ always() }}
         run: |
           $PATH_TO_GLUTEN_TE/$OS_IMAGE_NAME/gha/gha-checkout/clean.sh
+
+  upload-golden-file:
+    runs-on: velox-self-hosted
+    needs: [ubuntu2004-test-spark32, ubuntu2004-test-spark33, ubuntu2004-test-spark34]
+    steps:
+      - name: Upload golden files
```
Comment (ulysses-you) [inline .github/workflows/velox_be.yml:624]: if: failure()

File: .github/workflows/velox_be.yml [hunk 3]
```diff
@@ -631,3 +616,17 @@ jobs:
...
+        uses: actions/upload-artifact@v4
+        with:
+          name: golden-files-spark32
+          path: |
+            /tmp/${{ github.run_id }}/spark32/tpch-approved-plan/**
+      - name: Clean temp golden files
+        run: |
```
Comment (ulysses-you) [inline .github/workflows/velox_be.yml:631]: ditto

File: .github/workflows/velox_be.yml (hunk 2 above)
Comment (zwangsheng) [reply .github/workflows/velox_be.yml:656]: Right, we should trigger this upload job when the previous job fail.

And according to [Github Action Doc](https://docs.github.com/en/actions/learn-github-actions/expressions#failure), we should add this condition on job instead of adding on each step.

**Comment by ulysses-you** [comment]
@zwangsheng  is there any block on this pr ?

**Comment by zwangsheng** [comment]
> @zwangsheng is there any block on this pr ?

I'm still testing merge upload step. But IMO, we can leave merge step job in the following PR. WDYT @ulysses-you @PHILO-HE

**Comment by PHILO-HE** [comment]
> > @zwangsheng is there any block on this pr ?
> 
> I'm still testing merge upload step. But IMO, we can leave merge step job in the following PR. WDYT @ulysses-you @PHILO-HE

OK to me to separate the work into two or more PRs.
Assume the uploading action can also work well in the new CI we are working on. cc @zhouyuan.

**Comment by ulysses-you** [comment]
I'm fine to seperate pr. @PHILO-HE  IIUC the current action should work, just some code cleanup leave to another pr.

**Comment by zwangsheng** [comment]
Thanks for both @ulysses-you @PHILO-HE, i will revert this commit to focus on Spark 34 Golden Files, after some test, will turn this PR ready.

**Review by ulysses-you — APPROVED** [review]

File: .github/workflows/velox_be.yml [hunk 4]
```diff
@@ -670,3 +673,12 @@ jobs:
         if: ${{ always() }}
         run: |
           $PATH_TO_GLUTEN_TE/$OS_IMAGE_NAME/gha/gha-checkout/clean.sh
+
+  clean-up-tmp:
```
Comment (ulysses-you) [inline .github/workflows/velox_be.yml:677]: @zwangsheng  It seems a bit overkill to add a common job to do cleanup. The github action become complex.. I prefer to revert this change.

image

Comment (zwangsheng) [reply .github/workflows/velox_be.yml:677]: Well, if we want to merge upload action steps, here always will be a new job.

Comment (zwangsheng) [reply .github/workflows/velox_be.yml:677]: IMO, it is acceptable to add a new job to do the cleanup, especially since we will later incorporate the upload operations into this job.

Comment (ulysses-you) [reply .github/workflows/velox_be.yml:677]: I think we should merge upload and cleanup into one job and it seems unnecessary to pull out upload to a new job. There exists many similar jobs in gluten CI, if we want to make it clear, we should consider other jobs.
